From: S-P Chan <shihping.chan@gmail.com>
Date: Sun, 11 Feb 2024 12:14:19 +0800
Subject: tls: restore default to bypass thread guards

- restore <= 5.7.3 behaviour
- require user to opt-in to libssl thread-guards
  with tls_threads_mode = 1|2
---
 src/modules/tls/tls_mod.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/tls/tls_mod.c b/src/modules/tls/tls_mod.c
index beaf1b7..3359aaf 100644
--- a/src/modules/tls/tls_mod.c
+++ b/src/modules/tls/tls_mod.c
@@ -451,9 +451,9 @@ static int mod_child(int rank)
 #if OPENSSL_VERSION_NUMBER >= 0x010101000L
         /*
          * OpenSSL 3.x/1.1.1: create shared SSL_CTX* in worker to avoid init of
-         * libssl in rank 0(thread#1)
+         * libssl in rank 0(thread#1). Requires tls_threads_mode = 1 config.
          */
-        if(rank == PROC_SIPINIT) {
+        if((rank == PROC_SIPINIT && ksr_tls_threads_mode) || (rank == PROC_INIT && !ksr_tls_threads_mode)) {
 #else
         if(rank == PROC_INIT) {
 #endif
